common/src/main/java/net/neoforged/gradle/common/CommonProjectPlugin.java

```diff
@@ -374,6 +378,30 @@ private void applyAfterEvaluate(final Project project) {
                     definitionSet.forEach((identifier, definition) -> {
                         definition.configureRun(runImpl);
                     });
+
+                    //Handle dev login.
+                    final DevLogin devLogin = project.getExtensions().getByType(Subsystems.class).getDevLogin();
+                    final Tools tools = project.getExtensions().getByType(Subsystems.class).getTools();
+                    if (devLogin.getEnabled().get()) {
+                        //Dev login is only supported on the client side
+                        if (runImpl.getIsClient().get() && runImpl.getUseDevLogin().get()) {
+                            final String mainClass = runImpl.getMainClass().get();
+
+                            //We add the dev login tool to the runtime only configuration, of the first source set, this should suffice
+                            final SourceSet defaultSourceSet = runImpl.getModSources().get().get(0);
+                            final Configuration defaultRuntimeOnlyConfiguration = project.getConfigurations().maybeCreate(defaultSourceSet.getRuntimeOnlyConfigurationName());
```

Comment:
`runtimeOnly` is a leaking configuration, please user another one that inherits from `runtimeClasspath`.